roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -117,13 +117,14 @@ def run(self):
 
         if unavailable_images:
             unreachable = [reg for reg, reachable in self.reachable_registries.items() if not reachable]
+            if six.PY2:
+                unreachable = [reg.encode('utf8') for reg in unreachable]
+                self.registries["blocked"] = [reg.encode('utf8') for reg in self.registries["blocked"]]
+                unavailable_images = [reg.encode('utf8') for reg in unavailable_images]
+                self.registries["configured"] = [reg.encode('utf8') for reg in self.registries["configured"]]
             unreachable_msg = "Failed connecting to: {}\n".format(", ".join(unreachable))
             blocked_msg = "Blocked registries: {}\n".format(", ".join(self.registries["blocked"]))
             missing = ",\n    ".join(sorted(unavailable_images))
-            if six.PY2:
-                unreachable_msg = unreachable_msg.encode('utf8')
-                blocked_msg = blocked_msg.encode('utf8')
-                missing = missing.encode('utf8')
 
             msg = (
```

Comment:
This is the line that is throwing the error.  Python2 will automatically type the strings as necessary to unicode if they container those characters (or at least dynamically convert them).

msg is setup using default string type in python2, so it can't use .format() to put any unicode bits in.

I think if you change the lines in msg to u"something ..." it should work for every case.  We can get rid of the six.PY bits.

Reply:
Good idea, removed most of the `.encode` calls, looks much cleaner now

It seems `msg` still needs to be encoded (as other parts are using `str(msg)`)